From 185122b953ad990083d8748ef336b89205b64efc Mon Sep 17 00:00:00 2001
From: Daiki Ueno <ueno@unixuser.org>
Date: Mon, 9 Jun 2014 11:08:50 +0900
Subject: [PATCH] Don't send update-property signal until properties are
 registered

To reduce the number of D-Bus signals sent on focus-in.
---
 src/engine.vala | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/src/engine.vala b/src/engine.vala
index 7509e61..887e679 100644
--- a/src/engine.vala
+++ b/src/engine.vala
@@ -41,6 +41,7 @@ class KkcEngine : IBus.Engine {
     IBus.Keymap keymap;
     IBus.Property input_mode_prop;
     IBus.PropList prop_list;
+    bool properties_registered = false;
 
     Map<Kkc.InputMode, IBus.Property> input_mode_props =
         new HashMap<Kkc.InputMode, IBus.Property> ();
@@ -289,7 +290,8 @@ class KkcEngine : IBus.Engine {
                     prop.set_state (IBus.PropState.CHECKED);
                 else
                     prop.set_state (IBus.PropState.UNCHECKED);
-                update_property (prop);
+                if (properties_registered)
+                    update_property (prop);
             } while (iter.next ());
         }
 
@@ -304,7 +306,8 @@ class KkcEngine : IBus.Engine {
 #else
         input_mode_prop.set_label (symbol);
 #endif
-        update_property (input_mode_prop);
+        if (properties_registered)
+            update_property (input_mode_prop);
     }
 
     static void reload_dictionaries () {
@@ -550,8 +553,9 @@ class KkcEngine : IBus.Engine {
     }
 
     public override void focus_in () {
-        register_properties (prop_list);
         update_input_mode ();
+        register_properties (prop_list);
+        properties_registered = true;
         base.focus_in ();
     }
 
@@ -559,6 +563,7 @@ class KkcEngine : IBus.Engine {
         context.reset ();
         hide_preedit_text ();
         hide_lookup_table ();
+        properties_registered = false;
         base.focus_out ();
     }
 
-- 
1.9.0

